File: web/src/api/feedback.ts
import { http } from './client'

export type FeedbackCategory = 'bug' | 'suggestion' | 'appeal'
export type FeedbackStatus = 'PENDING' | 'PROCESSING' | 'REPLIED' | 'IGNORED'

export interface FeedbackItem {
  id: number
  userId?: number | null
  category: FeedbackCategory
  taskId?: number | null
  content: string
  contact?: string | null
  status: FeedbackStatus
  handledBy?: number | null
  handledReply?: string | null
  handledAt?: string | null
  createdAt: string
}

export interface SubmitFeedbackPayload {
  category: FeedbackCategory
  content: string
  taskId?: number
  contact?: string
  /** W3.c 登录接入前暂从上下文透传，登录接入后后端从 Sa-Token 取 */
  userId?: number
}

/** C 端提交反馈 */
export async function submitFeedback(payload: SubmitFeedbackPayload): Promise<{ id: number }> {
  const resp = await http.post('/api/v1/feedback', payload)
  return resp.data.data
}

/** C 端查自己的反馈历史 */
export async function listMyFeedback(userId: number): Promise<FeedbackItem[]> {
  const resp = await http.get('/api/v1/feedback/mine', { params: { userId } })
  return resp.data.data
}

/** 运营后台列表（W3.e 后台页面调用） */
export async function listFeedbackAdmin(params: {
  status?: FeedbackStatus | ''
  keyword?: string
  pageNum?: number
  pageSize?: number
} = {}): Promise<{ total: number; rows: FeedbackItem[] }> {
  const resp = await http.get('/admin/feedback/list', {
    params: { pageNum: 1, pageSize: 20, ...params },
  })
  return resp.data.data
}

/** 运营后台回复/处置 */
export async function handleFeedback(
  id: number,
  payload: { status?: FeedbackStatus; reply?: string; handledBy?: number },
): Promise<void> {
  await http.post(`/admin/feedback/${id}/handle`, payload)
}

File: backend-java/business-modules/ruoyi-detect/src/main/java/com/paperaigc/detect/domain/entity/Feedback.java
package com.paperaigc.detect.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 用户反馈实体 —— 对齐 scripts/patch-schema.sql · user_feedback
 *
 * <p>Phase 0 内存态，Phase B 接入 MyBatis-Plus 时补 @TableName / @TableId / @TableField 注解。</p>
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Feedback {

    /** 反馈 ID */
    private Long id;
    /** 提交用户 ID（Sa-Token 接入后从上下文取；未登录场景为 null） */
    private Long userId;
    /** 分类：bug / suggestion / appeal */
    private String category;
    /** 结果申诉时关联的检测任务 ID */
    private Long taskId;
    /** 反馈内容 ≤ 2000 字 */
    private String content;
    /** 联系方式（选填） */
    private String contact;
    /** 状态：PENDING / PROCESSING / REPLIED / IGNORED */
    private String status;
    /** 处理的运营账号 ID */
    private Long handledBy;
    /** 处理回复文本 */
    private String handledReply;
    /** 处理时间 */
    private LocalDateTime handledAt;
    /** 提交时间 */
    private LocalDateTime createdAt;
}
